flow/fuchsia/node.cc

```diff
@@ -0,0 +1,211 @@
+// Copyright 2017 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/fuchsia/node.h"
+#include <limits>
+#include "flutter/flow/fuchsia/session_transaction.h"
+#include "flutter/flow/matrix_decomposition.h"
+#include "third_party/skia/include/core/SkPath.h"
+
+namespace flow {
+
+Node::Node()
+    : resource_id_(std::numeric_limits<uint32_t>::max()),
+      transaction_(nullptr) {}
+
+Node::Node(Type type, uint32_t resource_id, SessionTransaction* transaction)
+    : resource_id_(resource_id), transaction_(transaction) {
+  if (!transaction_) {
+    return;
+  }
+  // Create the resource.
+  auto resource = mozart2::Resource::New();
+  switch (type) {
+    case Type::kEntity:
+      resource->set_entity_node(mozart2::EntityNode::New());
+      break;
+    case Type::kClip:
+      resource->set_clip_node(mozart2::ClipNode::New());
+      break;
+    case Type::kShape:
+      resource->set_shape_node(mozart2::ShapeNode::New());
+      break;
+  }
+
+  // Create the resource op.
+  auto create_resource = mozart2::CreateResourceOp::New();
```

Comment:
Boilerplate might be reduced by using session_helpers, or Jeff's higher-level API.  Is it kosher to use these within Flutter?